apps/web/src/components/ConnectWalletButton/CustomWalletAdvancedAddressDetails.tsx

```diff
@@ -0,0 +1,74 @@
+import { Name } from '@coinbase/onchainkit/identity';
+import { UserAvatar } from 'apps/web/src/components/ConnectWalletButton/UserAvatar';
+import classNames from 'classnames';
+import { useCallback, useState } from 'react';
+import { useAccount } from 'wagmi';
+
+export function CustomWalletAdvancedAddressDetails() {
+  const { address, chain } = useAccount();
+  const [copyText, setCopyText] = useState('Copy');
+
+  const handleCopyAddress = useCallback(() => {
+    try {
+      void navigator.clipboard.writeText(String(address));
+      setCopyText('Copied');
+      setTimeout(() => setCopyText('Copy'), 2000);
+    } catch (err) {
+      console.error('Failed to copy address:', err);
+      setCopyText('Failed to copy');
+      setTimeout(() => setCopyText('Copy'), 2000);
+    }
+  }, [address]);
```

Comment:
there are standard hooks for this you might consider that typically tend to have some nice edge case handling

https://usehooks-ts.com/react-hook/use-copy-to-clipboard

Reply:
updated in [861b859](https://github.com/base/web/pull/2003/commits/861b859ad6bcf4d4bffc399a955b68a9e8005d9a)